lib/Sema/CSDiagnostics.cpp

```diff
@@ -3399,6 +3402,38 @@ bool MissingMemberFailure::diagnoseForDynamicCallable() const {
   return false;
 }
 
+bool MissingMemberFailure::diagnoseForDefaultAnyInCollection() const {
+  auto &cs = getConstraintSystem();
+  auto *expr = getAsExpr(getAnchor());
+  auto *parentExpr = cs.getParentExpr(expr);
+  auto &solution = getSolution();
+
+  if (parentExpr && isa<UnresolvedMemberExpr>(expr)) {
+    auto parentType = resolveType(getType(parentExpr));
+
+    if (!cs.isCollectionType(parentType) && !parentType->is<TupleType>())
+      return false;
+
+    if (isa<TupleExpr>(parentExpr)) {
+      parentExpr = cs.getParentExpr(parentExpr);
+      if (!parentExpr)
+        return false;
+    }
+
+    if (auto *defaultableVar =
+            cs.getType(parentExpr)->getAs<TypeVariableType>()) {
+      if (llvm::any_of(
```

Comment:
Nit: `solution.DefaultedConstraints` is a set, so you could use `count()` instead of iteration here.